Upstream-Status: Backport

when install command sets the created directory mode, pseudo will change
the mode of the directory to 0700 incorrectly.

Signed-off-by: yanjun.zhu <yanjun.zhu@windriver.com>
Signed-off-by: Kai Kang <kai.kang@windriver.com>

--- a/ports/unix/guts/mkdirat.c
+++ b/ports/unix/guts/mkdirat.c
@@ -25,6 +25,7 @@
 		stat_rc = base_fstatat(dirfd, path, &buf, AT_SYMLINK_NOFOLLOW);
 #endif
 		if (stat_rc != -1) {
+			buf.st_mode = PSEUDO_DB_MODE(buf.st_mode, mode);
 			pseudo_client_op(OP_MKDIR, 0, -1, dirfd, path, &buf);
 		} else {
 			pseudo_debug(1, "mkdir of %s succeeded, but stat failed: %s\n",
